ibis/expr/datatypes/core.py

```diff
@@ -78,6 +78,198 @@ def castable(self, other):
 
         return castable(self, other)
 
+    @property
```

Comment:
I may prefer methods over properties, though I'm not sure. Do you have a strong preference?

Reply:
Either way we should update rlz.Shape enum to follow the same convention: https://github.com/ibis-project/ibis/blob/master/ibis/expr/rules.py#L32-L36

Reply:
I also lean slightly towards methods here, probably due to familiarity with the pyarrow `is_*` functions.

Reply:
Slight preference for properties, but I also don't feel too strong about it.

Reply:
Changed to methods.